src/ucp/core/ucp_worker.h

```diff
@@ -17,6 +17,13 @@
 #include <ucs/datastruct/queue_types.h>
 #include <ucs/async/async.h>
 
+
+/* The size of the private buffer in UCT descriptor headroom, which UCP may
+ * use for its own needs. This size does not include ucp_recv_desc_t length,
+ * because it is common for all cases and protocols (TAG, STREAM). */
+#define UCP_WORKER_HEADROOM_PRIV_SIZE 24
```

Comment:
minor: why 24? if this is a space for header, i would suggest to add static assertion for all protocols where rdesc is used.

Reply:
Now it is used for tag offload only. The assert is [here](https://github.com/openucx/ucx/pull/2485/files#diff-96a85f767c1bd283f2ba6265f1ca4a8eR732).